lib/resty/memcached.lua

```diff
@@ -736,4 +771,51 @@ function _M.close(self)
 end
 
 
+function _M.init_pipeline(self, n)
+    if self._reqs then 
+        return "already init pipeline"
+    end
+
+    if n and type(n) ~= 'number' then
+        return "buffer size is number type"
+    end
+    self._reqs = {
+        new_tab(n or 4, 0),
+        new_tab(n or 4, 0),
+    }
```

Comment:
I don't think it's a good idea to use three tables for the `_reqs`.
We can add another `_readers` field to save the readers, to save a table object.